nixos/modules/services/networking/headplane.nix

```diff
@@ -0,0 +1,108 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    attrsToList
+    listToAttrs
+    map
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    typeOf
+    types
+    ;
+  cfg = config.services.headplane;
+  settingsFormat = pkgs.formats.yaml { };
+  settingsFile = settingsFormat.generate "headplane-config.yaml" cfg.settings;
+  agentEnv = listToAttrs (
+    map (n: {
+      name = n.name;
+      value = if ((typeOf n.value) == "bool") then (if (n.value) then "true" else "false") else n.value;
+    }) (attrsToList cfg.agent.settings)
+  );
+in
+{
+  options.services.headplane = {
+    enable = mkEnableOption "headplane";
+    package = mkPackageOption pkgs "headplane" { };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
```

Comment:
The current configuration file doesn't seem too long. It would be nice to leverage the type system to help prevent users shooting themselves in the foot instead of a "free form" type.